libkernelflinger/security.c

```diff
@@ -216,6 +218,114 @@ struct rot_data_t* get_rot_data()
 	return &rot_data;
 }
 
+CHAR8* strrpl(CHAR8 *in, const CHAR8 src, const CHAR8 dst) 
+{
+    if (NULL == in)
+        return in;
+    CHAR8 *p = in;
+    while(*p != '\0'){
+        if(*p == src)
+            *p = dst;
+        p++;
+    }
+    return in;
+}
+
+static EFI_STATUS set_attestation_ids(UINT8 *src, UINT8 *cmpstr, UINT32 *idSize, UINT8 *id) 
+{
+    if ((src == NULL) || (cmpstr == NULL))
+        return EFI_INVALID_PARAMETER;
+    const char *delim2 = "=";
+    CHAR8 *savedPtr2;
+    const char d1 = ',';
+    const char d2 = ' ';
+    if (strncmp(src, cmpstr, strlen(cmpstr)) == 0) {
+        strtok_r(src, delim2, (char **)&savedPtr2);//Get the string after the equal sign
+        CHAR8* temp = strrpl(savedPtr2, d1, d2);//Converts comma to space in the string
+        *idSize = (strlen(temp)<ATTESTATION_ID_MAX_LENGTH)?strlen(temp):ATTESTATION_ID_MAX_LENGTH;
+        CopyMem(id, temp, *idSize);
+    }
+    return EFI_SUCCESS;
+}
+
+/* Update the struct attestation_ids for startup_information */
+EFI_STATUS update_attestation_ids(IN VOID *vendorbootimage)
+{
+    EFI_STATUS ret = EFI_SUCCESS;
+    if(vendorbootimage == NULL || ((struct vendor_boot_img_hdr_v3 *)vendorbootimage)->header_version < 4) {
+        init_attestation_ids();
```

Comment:
This function is already called previously in the efi_main, so I think it's not needed any more here.